packages/vulcan-forms/lib/components/FormComponent.jsx

```diff
@@ -1,44 +1,58 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { intlShape } from 'meteor/vulcan:i18n';
-import classNames from 'classnames';
 import { Components } from 'meteor/vulcan:core';
 import { registerComponent } from 'meteor/vulcan:core';
-import debounce from 'lodash.debounce';
 import get from 'lodash/get';
 import merge from 'lodash/merge';
+import isEqual from 'lodash/isEqual';
 import { isEmptyValue } from '../modules/utils.js';
 
 class FormComponent extends Component {
-  constructor(props) {
+  
+  constructor (props) {
     super(props);
 
-    const value = this.getValue(props);
+    this.state = {};
+  }
 
-    if (this.showCharsRemaining(props)) {
-      const characterCount = value ? value.length : 0;
-      this.state = {
-        charsRemaining: props.max - characterCount,
-      };
+  componentWillMount () {
+    if (this.showCharsRemaining()) {
+      const value = this.getValue();
+      this.updateCharacterCount(value);
     }
   }
 
-  // shouldComponentUpdate(nextProps, nextState) {
-  //   const { currentValues, deletedValues, errors } = nextProps;
-  //   const { path } = this.props;
-  //   const hasChanged = currentValues[path] && currentValues[path] !== this.props.currentValues[path];
-  //   const hasError = !!errors[path];
-  //   const hasBeenDeleted = deletedValues.includes(path) && !this.props.deletedValues.includes(path)
-  //   return hasChanged || hasError || hasBeenDeleted;
-  // }
-
+  shouldComponentUpdate (nextProps, nextState) {
+    // allow custom controls to determine if they should update
+    if (!['nested', 'number', 'url', 'email', 'textarea', 'checkbox',
```

Comment:
Hard-coding a list of field names like this seems weird to me. We could at least abstract it out in a `isCustomInput()` function

Reply:
Yes, I agree, that's ugly, I'll create `isCustomInput()`. Inside isCustomInput, could I use `type.toLowerCase() !== type` because custom inputs always start with a capital letter?

Reply:
Let's keep `isCustomInput()` like this for now since it's explicit at least. We'll figure out something more stable later.